atlasdb-client/src/main/java/com/palantir/atlasdb/off/heap/rocksdb/RocksDbPersistentTimestampStore.java

```diff
@@ -0,0 +1,133 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.off.heap.rocksdb;
+
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import org.rocksdb.ColumnFamilyDescriptor;
+import org.rocksdb.ColumnFamilyHandle;
+import org.rocksdb.RocksDB;
+import org.rocksdb.RocksDBException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.palantir.atlasdb.off.heap.ImmutableStoreNamespace;
+import com.palantir.atlasdb.off.heap.PersistentTimestampStore;
+import com.palantir.atlasdb.table.description.ValueType;
+import com.palantir.logsafe.Preconditions;
+import com.palantir.tracing.Tracers.ThrowingCallable;
+
+public final class RocksDbPersistentTimestampStore implements PersistentTimestampStore {
+    private static final Logger log = LoggerFactory.getLogger(RocksDbPersistentTimestampStore.class);
+
+    private final ConcurrentHashMap<UUID, ColumnFamilyHandle> availableColumnFamilies = new ConcurrentHashMap<>();
+    private final RocksDB rocksDB;
+
+    public RocksDbPersistentTimestampStore(RocksDB rocksDB) {
+        this.rocksDB = rocksDB;
+    }
+
+    @Override
+    @Nullable
+    public Long get(StoreNamespace storeNamespace, @Nonnull Long startTs) {
+        Preconditions.checkArgument(
+                availableColumnFamilies.containsKey(storeNamespace.uniqueName()),
+                "Store namespace does not exist");
+
+        byte[] byteKeyValue = ValueType.VAR_LONG.convertFromJava(startTs);
+        byte[] value = getWithExceptionHandling(availableColumnFamilies.get(storeNamespace.uniqueName()), byteKeyValue);
+
+        if (value == null) {
+            return null;
+        }
+        return startTs + (Long) ValueType.VAR_LONG.convertToJava(value, 0);
+    }
+
+    @Override
+    public void put(StoreNamespace storeNamespace, @Nonnull Long startTs, @Nonnull Long commitTs) {
+        Preconditions.checkArgument(
+                availableColumnFamilies.containsKey(storeNamespace.uniqueName()),
+                "Store namespace does not exist");
+
+        byte[] key = ValueType.VAR_LONG.convertFromJava(startTs);
+        byte[] value = ValueType.VAR_LONG.convertFromJava(commitTs - startTs);
+
+        putWithExceptionHandling(availableColumnFamilies.get(storeNamespace.uniqueName()), key, value);
+    }
+
+    @Override
+    public StoreNamespace createNamespace(@Nonnull String name) {
+        return ImmutableStoreNamespace.builder()
```

Comment:
nit: I'd suggest doing `createColumnFamily()` first for clarity, so it's more obvious that there are side effects

Reply:
Refactored.